pkg/collector/corechecks/snmp/internal/devicecheck/devicecheck.go

```diff
@@ -331,15 +330,15 @@ func (d *DeviceCheck) detectMetricsToMonitor(sess session.Session) error {
 		if err != nil {
 			return fmt.Errorf("failed to fetch sysobjectid: %s", err)
 		}
-		profile, err := profile.GetProfileForSysObjectID(d.config.Profiles, sysObjectID)
+		profile, err := d.config.Profiles.GetProfileNameForSysObjectID(sysObjectID)
```

Comment:
The behavior right now is:
1. If a profile name is hard-coded in the config, we set `.ProfileName` when the config is loaded
2. When we start looking for metrics, if there isn't a hard-coded profile, we fetch the sysobjectid and then set `.ProfileName` to the name of the profile that best matches the sysobjectid.
3. In both cases, after setting `.ProfileName`, we fetch the profile in question from the provider and recompute `config.OidConfig` based on it.

So if we want to fetch the profile directly instead of having the name as an intermediate step, we'll still need to handle the case where it's been hardcoded in the config.

Now that I think about it, though, it would be a good idea to do that - we shouldn't be caching the name of the best matching profile, because the best match could change when RC updates.